drivers/flash/flash_npcx_fiu_nor.c

```diff
@@ -0,0 +1,550 @@
+{
+	ARG_UNUSED(dev);
+
+	return &flash_npcx_parameters;
+};
+
+#ifdef CONFIG_FLASH_EX_OP_ENABLED
+static int flash_npcx_nor_ex_exec_uma(const struct device *dev, const uintptr_t in,
+					   void *out)
+{
+	struct npcx_ex_ops_uma_in *op_in =
+		(struct npcx_ex_ops_uma_in *)in;
+	const struct npcx_ex_ops_uma_out *op_out =
+		(const struct npcx_ex_ops_uma_out *)out;
+	int flag = 0;
+	struct npcx_uma_cfg cfg;
+
+	if (op_in == NULL) {
+		return -EINVAL;
+	}
+
+	/* Organize a UMA transaction */
+	cfg.opcode = op_in->opcode;
+	if (op_in->tx_count != 0) {
+		cfg.tx_buf = op_in->tx_buf;
+		cfg.tx_count = op_in->tx_count;
+		flag |= NPCX_UMA_ACCESS_WRITE;
+	}
+
+	if (op_in->addr_count != 0) {
+		cfg.addr.u32 = sys_cpu_to_be32(op_in->addr);
+		flag |= NPCX_UMA_ACCESS_ADDR;
+	}
+
+	if (op_out != NULL && op_out->rx_count != 0) {
+		cfg.rx_buf = op_out->rx_buf;
+		cfg.rx_count = op_out->rx_count;
+		flag |= NPCX_UMA_ACCESS_READ;
+	}
+
+	return flash_npcx_uma_transceive(dev, &cfg, flag);
+}
+
+static int flash_npcx_nor_ex_set_spi_spec(const struct device *dev, const uintptr_t in)
+{
+	struct npcx_ex_ops_qspi_oper_in *op_in =
+		(struct npcx_ex_ops_qspi_oper_in *)in;
+	struct flash_npcx_nor_data *data = dev->data;
+
+	/* Cannot disable write protection of internal flash */
+	if ((data->operation & NPCX_EX_OP_INT_FLASH_WP) != 0) {
+		if ((op_in->mask & NPCX_EX_OP_INT_FLASH_WP) != 0 && !op_in->enable) {
+			op_in->mask &= ~NPCX_EX_OP_INT_FLASH_WP;
+		}
+	}
+
+	if (op_in->enable) {
+		data->operation |= op_in->mask;
+	} else {
+		data->operation &= ~op_in->mask;
+	}
+
+	return 0;
+}
+
+static int flash_npcx_nor_ex_get_spi_spec(const struct device *dev, void *out)
+{
+	struct npcx_ex_ops_qspi_oper_out *op_out =
+		(struct npcx_ex_ops_qspi_oper_out *)out;
+	struct flash_npcx_nor_data *data = dev->data;
+
+	op_out->oper = data->operation;
+	return 0;
+}
+
+static int flash_npcx_nor_ex_op(const struct device *dev, uint16_t code,
+				const uintptr_t in, void *out)
+{
+	int ret = -ENOTSUP;
+
+	switch (code) {
+	case FLASH_NPCX_EX_OP_EXEC_UMA:
+		ret = flash_npcx_nor_ex_exec_uma(dev, in, out);
+		break;
+	case FLASH_NPCX_EX_OP_SET_QSPI_OPER:
+		ret = flash_npcx_nor_ex_set_spi_spec(dev, in);
+		break;
+	case FLASH_NPCX_EX_OP_GET_QSPI_OPER:
+		ret = flash_npcx_nor_ex_get_spi_spec(dev, out);
+		break;
+	}
+
+	return ret;
+}
+#endif
+
+static const struct flash_driver_api flash_npcx_nor_driver_api = {
+	.read = flash_npcx_nor_read,
+	.write = flash_npcx_nor_write,
+	.erase = flash_npcx_nor_erase,
+	.get_parameters = flash_npcx_nor_get_parameters,
+#if defined(CONFIG_FLASH_PAGE_LAYOUT)
+	.page_layout = flash_npcx_nor_pages_layout,
+#endif
+#if defined(CONFIG_FLASH_JESD216_API)
+	.sfdp_read = flash_npcx_nor_read_sfdp,
+	.read_jedec_id = flash_npcx_nor_read_jedec_id,
+#endif
+#ifdef CONFIG_FLASH_EX_OP_ENABLED
+	.ex_op = flash_npcx_nor_ex_op,
+#endif
+};
+
+static int flash_npcx_nor_init(const struct device *dev)
+{
+	if (IS_ENABLED(CONFIG_FLASH_NPCX_FIU_NOR_INIT)) {
```

Comment:
Could you invert this condition? Saves one indentation level on the rest of the function.

Reply:
Done.